src/transformers/models/imagebind/modeling_imagebind.py

```diff
@@ -0,0 +1,2101 @@
+        output_attentions = output_attentions if output_attentions is not None else self.config.output_attentions
+        output_hidden_states = (
+            output_hidden_states if output_hidden_states is not None else self.config.output_hidden_states
+        )
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
+        text_outputs = self.text_model(
+            input_ids=input_ids,
+            attention_mask=attention_mask,
+            position_ids=position_ids,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_dict=return_dict,
+        )
+
+        pooled_output = text_outputs[1]
+        text_features = self.text_projection(pooled_output)
+
+        return text_features
+
+    @add_start_docstrings_to_model_forward(IMAGEBIND_VISION_INPUTS_DOCSTRING)
+    def get_image_features(
+        self,
+        pixel_values: Optional[torch.FloatTensor] = None,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+    ) -> torch.FloatTensor:
+        r"""
+        Returns:
+            image_features (`torch.FloatTensor` of shape `(batch_size, output_dim`): The image embeddings obtained by
+            applying the projection layer to the pooled output of [`ImageBindVisionModel`].
+
+        Examples:
+
+        ```python
+        >>> from PIL import Image
+        >>> import requests
+        >>> from transformers import AutoProcessor, ImageBindModel
+
+        >>> model = ImageBindModel.from_pretrained("EduardoPacheco/imagebind-huge")
+        >>> processor = AutoProcessor.from_pretrained("EduardoPacheco/imagebind-huge")
+
+        >>> url = "http://images.cocodataset.org/val2017/000000039769.jpg"
+        >>> image = Image.open(requests.get(url, stream=True).raw)
+
+        >>> inputs = processor(images=image, return_tensors="pt")
+
+        >>> image_features = model.get_image_features(**inputs)
+        ```"""
+        # Use ImageBind model's config for some fields (if specified) instead of those in the vision components.
+        output_attentions = output_attentions if output_attentions is not None else self.config.output_attentions
+        output_hidden_states = (
+            output_hidden_states if output_hidden_states is not None else self.config.output_hidden_states
+        )
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+
+        batch_size = pixel_values.shape[0]
+
+        vision_outputs = self.vision_model(
+            pixel_values=pixel_values,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_dict=return_dict,
+        )
+
+        pooled_output = vision_outputs[1]  # pooled_output
+        image_features = self.vision_projection(pooled_output)
+
+        if pixel_values.ndim >= 5:
```

Comment:
same here, can we why?

Reply:
I've added comments explaining what is this in this commit https://github.com/huggingface/transformers/pull/30690/commits/53683a4c96f136eb16b7f1896af13a1ce0e85072

Audio and video inputs can be clipped (one audio can be chunked and one video we can take multiple clips) this is how the [original implementation](https://github.com/facebookresearch/ImageBind/blob/main/imagebind/models/imagebind_model.py#L470) dealt 
with the extra clips/chunks dimension.

We still need to add the video preprocessing for `ImageBind` should that be added in the `ImageBindImageProcessor` or should we have a different class for the video processing? IMO the latter seems better for avoiding complexity

Reply:
Thanks! And yes, it seems better to add another class for video processing, better separation of concerns